File: backend/src/main/java/com/herd/squire/models/creatures/companions/CompanionListObject.java
package com.herd.squire.models.creatures.companions;

import com.herd.squire.models.ListObject;
import com.herd.squire.models.creatures.CreatureHealth;

public class CompanionListObject extends ListObject {
    private CompanionType companionType;
    private CreatureHealth creatureHealth;
    private int maxHp;

    public CompanionListObject() {
        super();
    }

    public CompanionListObject(String id, String name, int sid, boolean author, CompanionType companionType,
                               CreatureHealth creatureHealth, int maxHp) {
        super(id, name, sid, author);
        this.companionType = companionType;
        this.creatureHealth = creatureHealth;
        this.maxHp = maxHp;
    }

    public CompanionType getCompanionType() {
        return companionType;
    }

    public void setCompanionType(CompanionType companionType) {
        this.companionType = companionType;
    }

    public CreatureHealth getCreatureHealth() {
        return creatureHealth;
    }

    public void setCreatureHealth(CreatureHealth creatureHealth) {
        this.creatureHealth = creatureHealth;
    }

    public int getMaxHp() {
        return maxHp;
    }

    public void setMaxHp(int maxHp) {
        this.maxHp = maxHp;
    }
}

File: frontend/src/app/shared/models/creatures/companions/companion-list-object.ts
import {ListObject} from '../../list-object';
import {CompanionType} from './companion-type.enum';
import {CreatureHealth} from '../creature-health';

export class CompanionListObject extends ListObject {
  companionType: CompanionType;
  creatureHealth: CreatureHealth;
  maxHp = 0;

  constructor(id: string, name: string, companionType: CompanionType, creatureHealth: CreatureHealth, maxHp: number) {
    super(id, name);
    this.companionType = companionType;
    this.creatureHealth = creatureHealth;
    this.maxHp = maxHp;
  }
}
